src/Tickets/Commerce/Gateways/Stripe/Client.php

```diff
@@ -232,9 +247,7 @@ public function get_payment_intent_transient() {
 	 * @param array $payment_intent payment intent data from Stripe
 	 */
 	public function store_payment_intent( $payment_intent ) {
-		if ( ! empty( $payment_intent['client_secret'] ) ) {
-			set_transient( $this->get_payment_intent_transient_name(), $payment_intent, 6 * HOUR_IN_SECONDS );
-		}
+		set_transient( $this->get_payment_intent_transient_name(), $payment_intent, 6 * HOUR_IN_SECONDS );
```

Comment:
I am confused, i though we spoke about moving this to the Order so we can avoid saving both the PaymentIntent and the Cart Hash. If we create an order as soon as we create the payment intent we shouldn't have a problem fetching the data using the ID.

Reply:
@bordoni you are correct, however, during implementation I faced an issue of Orders being too tightly coupled with the gateway provider - in that, to create an order without specifying a provider, we would need to refactor the way orders work. I think it's still worth pursuing, but not in the scope of this work.

Instead, what I did was create a payment intent on page-load and reload by the ID received from the front-end when creating the order, instead of relying only on the transients and naming structure on the back-end. This way, if the user pays with Stripe, they will always use the initial payment intent we created, so we won't be getting as many unused PIs as we were before. We still need a strategy to cancel these, but it's no longer as big of a deal as it was when we discussed this.